### DISCUSSION TITLE
Anchor link in routes

### DISCUSSION THREAD

**Discussion by markverg** [statamic/cms#6480] [Q&A]
Hi,

I'm creating pages with content blocks, using a hierchy collection like so:
- page1
  - contentblock1
  - contentblock2
- page2
  - contentblock1
  - contentblock2
 
When I create a link to a contentblock, I want it to look like this:
``https://siteurl/page1#contentblock2``. That works perfectly well for all menu's, as I can check the depth and add the # (hashtag) there. However, when I have our customer link to another page for instance when using the bard editor or a link field, I don't see how I can set-up the urls like that.

When defining a route for the pages collection I can't use the # (hashtag) there, it get's sanitized.
I even tried to insert it using Antlers, like so ``{{ parent_uri }}/{{ depth == 2 ? "#" : "" }}{{ slug }}``

What's the proper way of having Statamic be able to linking to an anchor tag within a page?
Or is there a completely different way I need to set-up the structure?

Many thanks!

**Comment by edalzell** [comment]
What is output with this? `{{ parent_uri }}/{{ depth == 2 ? "#" : "" }}{{ slug }}`?

**Reply by markverg** [reply]
> What is output with this? `{{ parent_uri }}/{{ depth == 2 ? "#" : "" }}{{ slug }}`?

It just ommits the hashtag so it then works out to be;
``https://siteurl/page1/contentblock2``

**Reply by edalzell** [reply]
what is the `depth` there?

**Reply by markverg** [reply]
Hi Erin,

page1 = depth 1
contentblock2 = depth 2

So this works:
``{{ parent_uri }}/{{ depth == 2 ? "-anchorlink-" : "" }}{{ slug }}``
and will output:
``https://siteurl/page1-anchorlink-contentblock2``

It's just when I put # there, the hashtag gets sanitized.
